packages/gatsby/src/commands/develop.js

```diff
@@ -91,6 +92,19 @@ async function startServer(program) {
       graphiql: true,
     })
   )
+
+  /**
+   * Refresh external data sources.
+   * If no GATSBY_REFRESH_TOKEN env var is available, then no Authorization header is required
+  **/
+  app.post(`/__refresh`, (req, res) => {
+    if (req.headers.authorization === process.env.GATSBY_REFRESH_TOKEN) {
```

Comment:
nitpick or misunderstanding perhaps, but if `process.env.GATSBY_REFRESH_TOKEN` is undefined, shouldn't refresh happen regardless of what is passed in `res.headers.authorization`?

Reply:
Thanks for having a look! You point about the authorization headers makes sense. Also intend on adding appropriate documentation. Just wanted to handle any requested changes to the implementation details before writing it out (eg if the refresh token is defined as a command line argument when starting the server, rather than an env var).

Reply:
I think I'd personally vote for it to be an env var